colcon_cmake/environment/cmake_module_path.py

```diff
@@ -21,12 +21,17 @@ def create_environment_hooks(self, prefix_path, pkg_name):  # noqa: D102
         hooks = []
 
         logger.log(1, "checking '%s' for CMake module files" % prefix_path)
+        count = 0
         for dirpath, _, filenames in os.walk(str(prefix_path)):
             for filename in filenames:
                 if filename.startswith('Find') and filename.endswith('.cmake'):
+                    count += 1
                     hooks += create_environment_hook(
-                        'cmake_module_path', prefix_path / dirpath,
-                        pkg_name, 'CMAKE_MODULE_PATH', '', mode='prepend')
+                        'cmake_module_path' +
+                        (str(count) if count > 1 else ''),
```

Comment:
I don't fully understand why this needs to have a number appended every time, won't you generate a lot of redundant environment hooks if there are several find modules in the same location?

Reply:
The outer loop iterates over different directories and the inner loop iterates over the files in one directory. For each directory only one hook can be generated (hence a [break](https://github.com/colcon/colcon-cmake/pull/19/files#diff-90c1e311baa57d65b1dfab53555fef91R35) after the generation in the inner loop). For a different directory another hook can be generated though which would point to a different directory.

To make an example for a package `foo` providing CMake modules in two different locations there would now be two hooks which prepend two different directories to `CMAKE_MODULE_PATH`:

* hook `cmake_module_path.sh` would prepend `<prefix>/share/foo/some/location` (even if there are multiple CMake modules in this directory only one hook is generated)
* hook `cmake_module_path2.sh` would prepend `<prefix>/share/foo/other/location`

Does that make sense?

Reply:
Yes 👍